vocode/streaming/models/agent.py

```diff
@@ -130,6 +130,13 @@ class AnthropicAgentConfig(AgentConfig, type=AgentType.ANTHROPIC.value):  # type
     model_name: str = CHAT_ANTHROPIC_DEFAULT_MODEL_NAME
     max_tokens: int = LLM_AGENT_DEFAULT_MAX_TOKENS
     temperature: float = LLM_AGENT_DEFAULT_TEMPERATURE
+    
+class LangchainAgentConfig(AgentConfig, type=AgentType.CHAT_GPT.value):  # type: ignore
+    prompt_preamble: str
+    model_name: str
+    provider: str
```

Comment:
Decided not to make an enum if langchain adds more providers to their factory method, and especially if users want to add custom chains/providers (based on config fields) since they'll probably utilize these fields